SConstruct

```diff
@@ -181,18 +180,41 @@ for k in platform_opts.keys():
     for o in opt_list:
         opts.Add(o)
 
-for x in module_list:
-    module_enabled = True
-    tmppath = "./modules/" + x
-    sys.path.insert(0, tmppath)
+# Detect modules.
+modules_detected = {}
+module_search_paths = ["modules"]  # Built-in path.
+
+if ARGUMENTS.get("custom_modules"):
+    paths = ARGUMENTS.get("custom_modules").split(",")
```

Comment:
Would be nice to *actually* fetch `custom_modules` via `custom.py` as described in #38821, but none of my attempts worked:
```python
custom_modules = ARGUMENTS.get("custom_modules")
if not custom_modules:
    # Assuming the option is set via a file instead.
    custom_modules = opts.args.get("custom_modules", "") # Returns empty string, but defined in `custom.py`...
```

It kinda works if we update the construction environment with those options earlier, but that means calling `opts.Update(env_base)` twice. Then it can be changed to:

```python
custom_modules = env_base["custom_modules"]
```